src/components/SearchInput/SearchInput.js

```diff
@@ -1,39 +1,17 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import { cx } from 'react-emotion';
 
-import Input from '../Input';
-import IconInputWrapper from '../IconInputWrapper';
+import IconInput from '../IconInput';
 import SearchIcon from './search.svg';
 
 /**
  * SearchInput component for forms.
  */
-const SearchInput = ({ disabled, ...props }) => (
-  <IconInputWrapper
-    role="search"
-    icon={({ className, disabledClassName }) => (
-      <SearchIcon
-        {...{ disabled }}
-        className={cx(className, { [disabledClassName]: disabled })}
-      />
-    )}
-    input={({ className }) => <Input {...{ ...props, disabled, className }} />}
-  />
+const SearchInput = ({ ...props }) => (
+  <IconInput role="search" {...props}>
+    {({ className }) => <SearchIcon {...{ className }} />}
```

Comment:
Hm yeah, so now the icon is not going to get a disabled style, no? So no `0.4` opacity when the input is disabled?

Reply:
See my comment above :)